### PR TITLE
Document Python build requirements

### PR DESCRIPTION
**Pull request by vstinner**
* Autoconf and aclocal: issue GH-89886
* SQLite 3.15.2: issue GH-105875
* Tcl/Tk 8.5.12: issue GH-91152

----
:books: Documentation preview :books:: https://cpython-previews--108646.org.readthedocs.build/

* https://cpython-previews--108646.org.readthedocs.build/en/108646/whatsnew/3.13.html
* https://cpython-previews--108646.org.readthedocs.build/en/108646/using/configure.html

### CODE REVIEW

**Comment by vstinner** [comment]
* @erlend-aasland: Please check the Autoconf, aclocal and SQLite minimum versions. -- I copied versions from What's New in Python 3.13
* @serhiy-storchaka: Please check the Tcl/Tk minimum version.

File: Doc/whatsnew/3.13.rst [hunk 1]
```diff
@@ -815,7 +815,7 @@ Build Changes
 =============
 
 * Autoconf 2.71 and aclocal 1.16.4 is now required to regenerate
-  :file:`!configure`.
+  the ``configure`` script.
```
Comment (AA-Turner) [inline Doc/whatsnew/3.13.rst]: No reason to remove the markup here, AFAICT:
```suggestion
  the :file:`configure` script.
```

Comment (vstinner) [reply Doc/whatsnew/3.13.rst]: Check how it is rendered: with the exclamation mark.

Comment (AA-Turner) [reply Doc/whatsnew/3.13.rst]: Ahh! No cross-reference is generated, so we should just remove the ``!``. Updated the suggestion. Ref: https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-file

Comment (vstinner) [reply Doc/whatsnew/3.13.rst]: I modified my PR to use the ``:file:`configure` `` syntax

File: Doc/using/configure.rst [hunk 2]
```diff
@@ -18,13 +20,19 @@ Features required to build CPython:
 
 * Support for threads.
 
-* OpenSSL 1.1.1 or newer for the :mod:`ssl` and :mod:`hashlib` modules.
+* OpenSSL 1.1.1 for the :mod:`ssl` and :mod:`hashlib` extension modules.
 
-* On Windows, Microsoft Visual Studio 2017 or later is required.
+* SQLite 3.15.2 for the :mod:`sqlite3` extension module.
```
Comment (erlend-aasland) [inline Doc/using/configure.rst:26]: You can add a `.. versionchanged:: 3.13` for the SQLite requirement.

File: Doc/using/configure.rst [hunk 3]
```diff
@@ -18,13 +20,19 @@ Features required to build CPython:
 
 * Support for threads.
 
-* OpenSSL 1.1.1 or newer for the :mod:`ssl` and :mod:`hashlib` modules.
+* OpenSSL 1.1.1 for the :mod:`ssl` and :mod:`hashlib` extension modules.
```
Comment (erlend-aasland) [inline Doc/using/configure.rst]: But the recommendation is minimum 3.0.9; it is not 1.1.1.

**Comment by vstinner** [comment]
@erlend-aasland: Please review the updated PR.

File: Doc/using/configure.rst [hunk 4]
```diff
@@ -18,13 +20,23 @@ Features required to build CPython:
 
 * Support for threads.
 
-* OpenSSL 1.1.1 or newer for the :mod:`ssl` and :mod:`hashlib` modules.
+* OpenSSL 1.1.1 is the minimum version and OpenSSL 3.0.9 is the recommended
```
Comment (AA-Turner) [inline Doc/using/configure.rst:23]: Should we capture that 3.0.9 is the recommended minimum, currently this sentence could be read as ==3.0.9, to my eye.

A

Comment (vstinner) [reply Doc/using/configure.rst:23]: Ok, I changed the phrasing.

**Review by erlend-aasland — APPROVED** [review]

**Review by serhiy-storchaka — COMMENTED** [review]
Tcl/Tk version 8.4 is required in 3.5.

Tcl/Tk version 8.3.1 is required in 3.1.

**Comment by serhiy-storchaka** [comment]
The earliest `versionchanged` here is for 3.5, so I think we can omit 3.1, but add to 3.5.

**Comment by vstinner** [comment]
> Tcl/Tk version 8.4 is required in 3.5.
> Tcl/Tk version 8.3.1 is required in 3.1.

Good to know. I completed the doc.

@serhiy-storchaka: Would you mind to review the updated PR?

**Comment by vstinner** [comment]
> 3.10: Require SQLite 3.7.15

Added.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. Seems that other extension modules either shipped with some default version (like zlib or decimal), or do not have explicit version requirements.

**Comment by vstinner** [comment]
Merged, I added Erlend and Serhiy as co-authors of the PR :-)

**Pull request merged**